src/fty_discovery_server.cc

```diff
@@ -687,7 +694,16 @@ void ftydiscovery_create_asset(fty_discovery_server_t* self, zmsg_t** msg_p)
                     link.dest = fty::convert<uint32_t>(DBAssets::name_to_asset_id(iname));
                 }
                 auto conn = tntdb::connectCached(DBConn::url);
-                DBAssetsInsert::insert_into_asset_links(conn, self->default_values_links);
+                // In device centric view for sts, add default link for the 2 inputs if default links have only one entry
+                if (self->device_centric && streq(name, "sts") && self->default_values_links.size() == 1) {
```

Comment:
This condition "self->device_centric && streq(name, "sts")" used at least twice... could you just declare bool = self->device_centric && streq(name, "sts")

"self->default_values_links.size()"
for containers is better to use empty() check

Reply:
I agree but in this case I need to know if there is exactly 1 element in the list (not empty and if > 1 no need to add another element)